app/components/UserValidator/index.tsx

```diff
@@ -125,29 +128,53 @@ const Validator = ({ handleSelect, validateAbakusGroup }: Props) => {
             if (!validateAbakusGroup || (isUser(user) && user.isAbakusMember)) {
               const sound = new window.Audio(goodSound);
               sound.play();
-              if (validateAbakusGroup) {
-                displayResult(`${username} er Abakus-medlem`, true);
-              } else {
-                displayResult(`${username} ble registrert`, true);
-              }
+
+              displayResult(
+                {
+                  message: validateAbakusGroup ? `${username} er Abakus-medlem` : `${username} ble registrert`,
+                  icon: <Check />,
+                  color: 'var(--success-color)',
+                },
+                true,
+              );
             } else {
-              displayResult(`${username} er ikke medlem av Abakus!`);
+              displayResult(
+                {
+                  message: `${username} er ikke medlem av Abakus`,
+                  icon: <X />,
+                  color: 'var(--danger-color)',
+                },
+                false,
+              );
             }
           },
           (err) => {
             setIsLoading(false);
             const payload = get(err, 'payload.response.jsonData');
-            if (payload && payload.errorCode === 'not_registered') {
-              displayResult(`${username} er ikke påmeldt arrangementet`);
-            } else if (payload && payload.errorCode === 'already_present') {
-              displayResult(`${username} er allerede registrert`);
-            } else if (payload && payload.detail === 'Not found.') {
-              displayResult(`Brukeren finnes ikke!\nBrukernavn: ${username}`);
-            } else {
-              displayResult(
-                `Det oppsto en uventet feil: ${JSON.stringify(payload || err)}`,
-              );
-            }
+
+            const errorMessages = {
+              not_registered: `${username} er ikke påmeldt arrangementet`,
+              already_present: `${username} er allerede registrert`,
+              unregistered: `${username} har meldt seg av`,
+              not_properly_registered: `${username} sin påmelding er i limbo. Ta kontakt med Webkom`,
+              waitlisted: `${username} er på venteliste`,
+              late_or_absent: `${username} har blitt registrert som ikke tilstede`,
+              missing_payment: `${username} har ikke betalt`,
+              no_user: `Brukeren finnes ikke! Brukernavn: ${username}`,
+            };
+
+            const errorMessage =
+              (payload && errorMessages[payload.errorCode]) ??
```

Comment:
I think this would crash the webapp if we make new codes

```suggestion
              (payload && payload.errorCode in errorMessages) ?  errorMessages[payload.errorCode] :
```